paywall/src/data-iframe/cache/localStorage.js

```diff
@@ -86,6 +124,58 @@ export async function put(window, networkId, accountAddress, type, value) {
   window.localStorage.setItem(key, JSON.stringify(container))
 }
 
+/**
+ * Retrieve the current cached account
+ *
+ * @param {object} window this is the global context, either global, window, or self
+ * @returns {string}
+ */
+export async function getAccount(window) {
+  if (!localStorageAvailable(window)) {
+    throw new Error('Cannot get value from localStorage')
+  }
+  return window.localStorage.getItem('__unlockProtocol.account') || null
+}
+
+/**
+ * Set the current cached account
+ *
+ * @param {object} window this is the global context, either global, window, or self
+ * @param {string} account the ethereum account address of the current user
+ */
+export async function setAccount(window, account) {
+  if (!localStorageAvailable(window)) {
+    throw new Error('Cannot put value into localStorage')
+  }
+  window.localStorage.setItem('__unlockProtocol.account', account)
+}
+
+/**
+ * Retrieve the current cached ethereum network
+ *
+ * @param {object} window this is the global context, either global, window, or self
+ * @returns {number}
+ */
+export async function getNetwork(window) {
+  if (!localStorageAvailable(window)) {
+    throw new Error('Cannot get value from localStorage')
+  }
+  return +window.localStorage.getItem('__unlockProtocol.network') || null
+}
+
+/**
+ * Set the current cached network
+ *
+ * @param {object} window this is the global context, either global, window, or self
+ * @param {number} network the id of the current ethereum network
+ */
+export async function setNetwork(window, network) {
```

Comment:
Not blocking but I think that we could probably generalize these getters and setters